bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrPlugin.kt

```diff
@@ -42,31 +43,70 @@ internal class AnrPlugin : Plugin {
 
     override fun unload() = disableAnrReporting()
 
+    private fun getNotifyAnrDetectedIndex(stackTrace: Array<StackTraceElement>): Int {
+        for (i in 0..stackTrace.size) {
+            val frame = stackTrace[i]
+            if (frame.methodName == "notifyAnrDetected" && frame.className == "com.bugsnag.android.AnrPlugin") {
+                return i
+            }
+        }
+        return -1
+    }
+
+    private fun isNativeANR(stackTrace: Array<StackTraceElement>): Boolean {
+        // True if the frame leading to "notifyAnrDetected" is native.
+        // This is not perfect, because some JVM calls are technically native (like sleep).
+
+        val notifyIndex = getNotifyAnrDetectedIndex(stackTrace)
+        if (notifyIndex < 0) {
+            return false
+        }
+
+        return stackTrace[notifyIndex + 1].isNativeMethod
+    }
+
+    private fun mergeTraces(jvmTrace: Array<StackTraceElement>, nativeTrace: List<Stackframe>):
+        Array<StackTraceElement> {
```

Comment:
This function needs to return `com.bugsnag.android.Stackframe` - it appears as though it's currently discarding the information gathered in the NDK stacktrace by attempting to convert it into a `StacktraceElement`

Reply:
The NDK is only filling out filename, method, and line number. I didn't realise when writing the native portion that the JVM side is taking StackTraceElements and then converting to Stackframe. Maybe I should modify the native side to return StackTraceElement instead since the other Stackframe fields are used only for react native, or for project packages?